.circleci/config.yml

```diff
@@ -68,6 +68,70 @@ jobs:
         make images-nodep
         make push-nodep
 
+  e2e:
+    machine: true
+    environment:
+      CHANGE_MINIKUBE_NONE_USER: true
+      KUBERNETES_VERSION: v1.10.0
+    steps:
+    - run:
+        name: Create directory structure
+        command: |
+          # Create directory
+          sudo mkdir -p /go/src/github.com/kubermatic/cluster-api-provider-digitalocean
+          # Set up permissions
+          sudo chown -R $USER:$USER /go
+    - restore_cache:
+        key: repo-{{ .Environment.CIRCLE_SHA1 }}
+    - run:
+        name: Install dependencies
+        command: |
+          # Download kubectl, which is a requirement for using minikube and running e2e tests.
+          curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+          # Download minikube.
+          curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+          # Download kubeadm.
+          curl -Lo kubeadm https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubeadm && chmod +x kubeadm && sudo mv kubeadm /usr/bin/
+    - run:
+        name: Set up minikube
+        command: |
+          # Make root rshared in order to fix kube-dns problems.
+          sudo mount --make-rshared /
+          # Start minikube.
+          sudo minikube start --vm-driver=none --bootstrapper=localkube --kubernetes-version=${KUBERNETES_VERSION} --feature-gates=CustomResourceSubresources=true --extra-config=apiserver.Authorization.Mode=RBAC
+          # Set correct permissions on minikube and kubeconfig files.
+          sudo chown -R $USER:$USER $HOME/.kube
```

Comment:
For some reason this is required, but it shouldn't be. To be debugged.

Reply:
When `CHANGE_MINIKUBE_NONE_USER` environment variable is set, Minikube should adjust permissions automatically, so non-root user owns it, but that's not happening here.

I will try some fix tomorrow.

Reply:
Resolved by running minikube start with `sudo -E`